File: medium/python/2933.py
# https://leetcode.com/problems/high-access-employees/description/

class Solution:
    def findHighAccessEmployees(self, access_times: List[List[str]]) -> List[str]:
        all = defaultdict(list)

        for time in access_times:
            all[time[0]].append(int(time[1][0:2]) * 60 + int(time[1][2:]))

        res = []

        for k, v in all.items():
            if len(v) < 3:
                continue
            v.sort()
            for i in range(2, len(v)):
                if v[i] - v[i - 2] < 60:
                    res.append(k)
                    break

        return res
        

File: medium/cpp/2933.cpp
// https://leetcode.com/problems/high-access-employees/description/

class Solution
{
public:
    std::vector<std::string> findHighAccessEmployees(std::vector<std::vector<std::string>>& access_times)
    {
        std::map<std::string, std::vector<int>> emp;

        for (const auto& at : access_times)
        {
            const auto ts = std::stoi(at[1].substr(0, 2)) * 60 + std::stoi(at[1].substr(2));
            emp[at[0]].push_back(ts);
        }

        std::vector<std::string> res;
        res.reserve(std::size(access_times));

        for (auto& e : emp)
        {
            if (std::size(e.second) < 3)
            {
                continue;
            }

            std::sort(std::begin(e.second), std::end(e.second));

            for (auto i = 2; i < std::size(e.second); i++)
            {
                if (e.second[i] - e.second[i - 2] < 60)
                {
                    res.push_back(e.first);
                    break;
                }
            }
        }

        return res;
    }
};